From 1f03944f3c5a687d3065c18e6d0debd7540cc7d6 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 22 Jun 2021 19:52:16 +0300
Subject: [PATCH] [SW-0] module_lkd: add state shift calls comments

It might be confusing why we do state shift also in case of errors, but
in fact logically it is safe, so to not stumble upon this next time -
add comments explaining why is it correct to call the
module_lkd_state_shift() function in any case.

Change-Id: I7b55d532127b7da1693f157caf95a74de520c9e7
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot/module_lkd.c b/preboot/module_lkd.c
index 1b04bbc..0c84bb0 100644
--- a/preboot/module_lkd.c
+++ b/preboot/module_lkd.c
@@ -355,6 +355,7 @@
 		module_lkd_send_status(COMMS_STS_ERR, COMMS_SRAM, 0);
 	}
 
+	/* no problem to shift on error, as later we check the validity */
 	module_lkd_state_shift();
 
 	preboot_puts_raw("Got host ");
@@ -525,6 +526,7 @@
 		preboot_puts(": default case - err\n");
 	}
 
+	/* no problem to shift here on error as it will only clear the do_cmd */
 	module_lkd_state_shift();
 }
 
